book/src/user_manual/examining_welcome_message.md

```diff
@@ -0,0 +1,49 @@
+# Examining a welcome message
+
+When a client is invited to join a group, the application can allow the client to decide whether or not to join the group. In order to determine whether to join the group, the application can inspect information provided in the welcome message, such as who invited them, who else is in the group, what extensions are available, and more. If the application decides not to join the group, data must be cleaned up in the storage provider.
```

Comment:
"data must be cleaned up in the storage provider." I think the application shouldn't have to worry about that level of detail. Instead, we should say something like "the welcome must be discarded to ensure that the local state is cleaned up".